packages/pubsub/test/connection-pool.js

```diff
@@ -347,6 +392,66 @@ describe('ConnectionPool', function() {
       pool.createConnection();
     });
 
+    describe('channel', function() {
+      var channelReadyEvent = 'channel.ready';
+      var channelErrorEvent = 'channel.error';
+
+      describe('error', function() {
+        it('should remove the channel ready event listener', function() {
+          pool.createConnection();
+          assert.strictEqual(pool.listenerCount(channelReadyEvent), 1);
+
+          pool.emit(channelErrorEvent);
+          assert.strictEqual(pool.listenerCount(channelReadyEvent), 0);
+        });
+
+        it('should cancel the connection', function() {
+          pool.createConnection();
+          pool.emit(channelErrorEvent);
+
+          assert.strictEqual(fakeConnection.canceled, true);
```

Comment:
`canceled` being set to `true` looks to be done somewhere else, not in the `createConnection` body of code. Can you just stub the `connection.cancel` method to check that it was called?

Reply:
Actually that's exactly what is happening here. https://github.com/callmehiphop/gcloud-node/blob/728575b113a3715e4f83554c4859c4ce7aed898f/packages/pubsub/test/connection-pool.js#L65